test/erlfmt_format_SUITE.erl

```diff
@@ -2574,13 +2574,47 @@ receive_expression(Config) when is_list(Config) ->
         "    2 -> two\n"
         "end\n"
     ),
-    % ?assertSame(
-    %     "receive\n"
-    %     "after\n"
-    %     "    % foo\n"
-    %     "    0 -> ok\n"
-    %     "end\n"
-    % ),
+    ?assertFormat(
+        "receive\n"
+        "after\n"
+        "    0 -> ok\n"
+        "    % receive post comment\n"
+        "end\n",
+        "receive\n"
+        "after 0 ->\n"
+        "    ok\n"
+        "    % receive post comment\n"
+        "end\n"
+    ),
+    ?assertFormat(
+        "receive\n"
+        "after\n"
+        "    % before zero\n"
+        "    0 -> ok\n"
+        "end\n",
+        "receive\n"
+        "after\n"
+        "% before zero\n"
+        "0 ->\n"
+        "    ok\n"
+        "end\n"
+    ),
+    ?assertFormat(
+        "receive\n"
+        "after\n"
+        "    %% foo\n"
+        "    0 ->\n"
+        "        \"abc\"\n"
+        "        \"def\"\n"
+        "end.\n",
+        "receive\n"
+        "after\n"
+        "%% foo\n"
```

Comment:
This is a downside of the current solution.
Where the clause is not indented.

The problem with indenting the clause is then when we loose the `0 ->` on the same line as the after.
```
?assertFormat(
        "receive\n"
        "    1 -> ok\n"
        "after 0 ->\n"
        "    ok\n"
        "end\n",
        "receive\n"
        "    1 -> ok\n"
        "after\n"
        "    0 ->\n"
        "        ok\n"
        "end\n"
    ),
```

Reply:
I am wondering if the current solution is an okay compromise, at least the comment stays in place.

Reply:
For reference this would be the code for the formatting in the top case
```
receive_after_to_algebra(Expr, [HBody | _] = Body) ->
    ExprD = expr_to_algebra(Expr),
    BodyD = block_to_algebra(Body),
    MaybeForce = maybe_force_breaks(has_break_between(Expr, HBody)),
    ArrowBody = group(nest(concat([<<"->">>, break(), MaybeForce, BodyD]), ?INDENT)),
    Doc = group(nest(concat([break(), ExprD, <<" ">>, ArrowBody]), ?INDENT, break)),
    concat(<<"after">>, Doc).
```

Reply:
The groups break upward, so all the nesting and breaks are triggered